specification/agfood/data-plane/Microsoft.AgFoodPlatform/preview/2021-03-31-preview/agfood.json

```diff
@@ -5558,28 +5562,72 @@
         }
       }
     },
-    "/oauth/tokens/:remove": {
+    "/oauth/tokens/:connect": {
       "post": {
         "tags": [
           "OAuthTokens"
         ],
-        "description": "Deletes OAuth Token for given oauth provider Id and farmer Id.",
-        "operationId": "OAuthTokens_Delete",
+        "description": "Returns Connection link needed in the OAuth flow.",
+        "operationId": "OAuthTokens_GetOAuthConnectionLink",
+        "consumes": [
+          "application/json"
+        ],
         "produces": [
           "application/json"
         ],
         "parameters": [
           {
             "in": "query",
-            "name": "farmerId",
-            "description": "Id of the associated farmer.",
+            "name": "api-version",
+            "description": "The requested API version",
             "required": true,
             "type": "string"
           },
           {
-            "in": "query",
-            "name": "oauthProviderId",
-            "description": "Id of the associated oauth provider.",
+            "in": "body",
+            "name": "connectRequest",
+            "description": "OAuth Connect Request.",
+            "schema": {
+              "$ref": "#/definitions/OAuthConnectRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type": "string"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "OAuthTokens_GetOAuthConnectionLink": {
+            "$ref": "./examples/OAuthTokens_GetOAuthConnectionLink.json"
+          }
+        }
+      }
+    },
+    "/oauth/tokens/remove/{jobId}": {
+      "get": {
+        "tags": [
+          "OAuthTokens"
+        ],
+        "description": "Get remove job for OAuth token.",
```

Comment:
Maybe something like "Get details on a specified cascade delete job for OAuth tokens."

Reply:
Updated.